From 1b4a5bf98911067c7f1dd504cda1c5722588467f Mon Sep 17 00:00:00 2001
Message-Id: <1b4a5bf98911067c7f1dd504cda1c5722588467f.1431284209.git.chang-joon.lee@intel.com>
In-Reply-To: <89460f3fc3d6c0659d44897a6adec3c3fecc5564.1431284209.git.chang-joon.lee@intel.com>
References: <89460f3fc3d6c0659d44897a6adec3c3fecc5564.1431284209.git.chang-joon.lee@intel.com>
From: John Harrison <John.C.Harrison@Intel.com>
Date: Wed, 8 Apr 2015 18:05:50 +0100
Subject: [PATCH 42/68] SQUASHME! [VPG]: drm/i915: Fix for leaking fence
 objects

In the case where something fails during the execbuff IOCTL, the
driver is supposed to clean up any outstanding items that got
allocated but are no longer required. This was not happening for the
native sync fence object.

This patch adds in the measing clean up code. It could be squashed
into the original patch that added the native sync support to the
execbuff code:
  commit 6b106574fa9863d9c4245b0a374fca50e09e1f2b
  Author: John Harrison <John.C.Harrison@Intel.com>
  Date:   Tue Apr 1 17:25:21 2014 +0100
  Subject: REVERTME [VPG]: drm/i915: Refactored native sync code for better file splits
  Change-Id: I8c2653787a1b15b4eb00c1dfa05b5e50ab4d2390

Change-Id: I777a739265f212f0fbf61c5968e669a46b7296f6
For: VIZ-1587
For: VIZ-4741
Signed-off-by: John Harrison <John.C.Harrison@Intel.com>
---
 drivers/gpu/drm/i915/i915_gem_execbuffer.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_gem_execbuffer.c b/drivers/gpu/drm/i915/i915_gem_execbuffer.c
index 2b682fe..577c1ae 100644
--- a/drivers/gpu/drm/i915/i915_gem_execbuffer.c
+++ b/drivers/gpu/drm/i915/i915_gem_execbuffer.c
@@ -1787,6 +1787,9 @@ err:
 			i915_gem_context_unreference(params->ctx);
 	}
 
+	if (params->fence_wait)
+		sync_fence_put(params->fence_wait);
+
 	/* Free the OLR again in case the failure occurred after it had been
 	 * allocated. */
 	i915_gem_request_assign(&ring->outstanding_lazy_request, NULL);
-- 
1.7.9.5

